core/services/relay/evm/dual_contract_transmitter.go

```diff
@@ -0,0 +1,181 @@
+
+	// Primary transmission
+	payload, err := oc.contractABI.Pack("transmit", rawReportCtx, []byte(report), rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "abi.Pack failed")
+	}
+
+	transactionErr := errors.Wrap(oc.transmitter.CreateEthTransaction(ctx, oc.contractAddress, payload, txMeta), "failed to send Eth transaction")
+
+	// Secondary transmission
+	secondaryPayload, err := oc.dualTransmissionABI.Pack("transmitSecondary", rawReportCtx, []byte(report), rs, ss, vs)
+	if err != nil {
+		return errors.Wrap(err, "transmitSecondary abi.Pack failed")
+	}
+
+	err = errors.Wrap(oc.transmitter.CreateSecondaryEthTransaction(ctx, secondaryPayload, txMeta), "failed to send secondary Eth transaction")
+
+	return errors2.Join(transactionErr, err)
+}
+
+// LatestConfigDigestAndEpoch retrieves the latest config digest and epoch from the OCR2 contract.
+// It is plugin independent, in particular avoids use of the plugin specific generated evm wrappers
+// by using the evm client Call directly for functions/events that are part of OCR2Abstract.
+func (oc *dualContractTransmitter) LatestConfigDigestAndEpoch(ctx context.Context) (ocrtypes.ConfigDigest, uint32, error) {
+	latestConfigDigestAndEpoch, err := callContract(ctx, oc.contractAddress, oc.contractABI, "latestConfigDigestAndEpoch", nil, oc.contractReader)
+	if err != nil {
+		return ocrtypes.ConfigDigest{}, 0, err
+	}
+	// Panic on these conversions erroring, would mean a broken contract.
+	scanLogs := *abi.ConvertType(latestConfigDigestAndEpoch[0], new(bool)).(*bool)
+	configDigest := *abi.ConvertType(latestConfigDigestAndEpoch[1], new([32]byte)).(*[32]byte)
+	epoch := *abi.ConvertType(latestConfigDigestAndEpoch[2], new(uint32)).(*uint32)
+	if !scanLogs {
+		return configDigest, epoch, nil
+	}
+
+	// Otherwise, we have to scan for the logs.
+	latest, err := oc.lp.LatestLogByEventSigWithConfs(ctx, oc.transmittedEventSig, oc.contractAddress, 1)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			// No transmissions yet
+			return configDigest, 0, nil
+		}
+		return ocrtypes.ConfigDigest{}, 0, err
+	}
+	return parseTransmitted(latest.Data)
+}
+
+// FromAccount returns the account from which the transmitter invokes the contract
+func (oc *dualContractTransmitter) FromAccount(ctx context.Context) (ocrtypes.Account, error) {
+	return ocrtypes.Account(oc.transmitter.FromAddress(ctx).String()), nil
+}
+
+func (oc *dualContractTransmitter) Start(ctx context.Context) error { return nil }
+func (oc *dualContractTransmitter) Close() error                    { return nil }
+
+// Has no state/lifecycle so it's always healthy and ready
+func (oc *dualContractTransmitter) Ready() error { return nil }
+func (oc *dualContractTransmitter) HealthReport() map[string]error {
+	return map[string]error{oc.Name(): nil}
+}
+func (oc *dualContractTransmitter) Name() string { return oc.lggr.Name() }
```

Comment:
Is there some reason that this type must implement `Service`?

Reply:
Yes, in order to satisfy the `evm.ContractTransmitter` interface it must implement `services.ServiceCtx`